hera_sim/visibilities/simulators.py

```diff
@@ -137,40 +163,59 @@ def convert_healpix_to_point_sources(hmap):
         return np.array([ra, dec, flux])
 
     def simulate(self):
+        """Perform the visibility simulation"""
+        self._write_history()
+        self._simulate()
+
+    def _simulate(self):
+        """Subclass-specific simulation method, to be overwritten."""
         pass
 
     @property
     def nside(self):
+        """Nside parameter of the sky healpix map"""
         if self.sky_intensity is not None:
-            return healpy.get_nside(self.sky_intensity)
+            return healpy.get_nside(self.sky_intensity[0])
         else:
             return self._nside
 
     @cached_property
-    def n_ant(self):
-        """Number of antennas in array"""
-        return self.antpos.shape[0]
+    def n_pix(self):
+        """Number of pixels in the sky map"""
+        return self.sky_intensity.shape[1]
 
     @cached_property
-    def n_lsts(self):
-        """Number of times (LSTs)"""
-        return self.lsts.shape[0]
+    def n_ant(self):
+        """Number of antennas in array"""
+        return self.uvdata.get_ants().shape[0]
 
     @cached_property
     def n_beams(self):
         """Number of beam models used."""
-        return self.beams.shape[0]
+        return len(self.beams)
 
-    @cached_property
-    def n_pix(self):
-        """Number of pixels in the sky map"""
-        return self.sky_intensity.size
+    def _write_history(self):
+        """
+        Write pertinent details of simulation to the UVData's history.
+        """
+        self.uvdata.history += "Visibility Simulation performed with hera_sim's {} simulator\n".format(
+            self.__class__.__name__)
+        self.uvdata.history += "Class Repr: {}".format(repr(self))
 
-    @cached_property
-    def n_pix_beam(self):
-        """Number of pixels in the beam maps"""
-        try:
-            return self.beams.shape[1]
-        except IndexError:
-            return 0
+    def write_config_file(self, **kwargs):
+        """
+        Write out a YAML config file corresponding to the current UVData
+        object.
 
+        Args:
+            **kwargs: any options are passed to :func:`uvdata_to_config_file`.
+        """
+        uvdata_to_config_file(self.uvdata, **kwargs)
```

Comment:
You may want to note that this will write out some dummy values for certain things that can't be specified by a UVData object. Specifically -- catalog info, paths to telescope configuration files (which includes antenna layouts and the beam dictionary) and the "filing" parameters (output filename/ etc.).

There is a separate function to write telescope configuration files from a given UVData object, `uvdata_to_telescope_config`.

Reply:
Thanks. I've change to using ``uvdata_to_telescop_config`` in 9abbd6f53885fa7e32b89ddbaa7acbbc60de3bf6

Reply:
by the way, I think I should point out that I find the reading/writing of config files in pyuvsim to be hard to follow. It could be a lot more modular (like how about a separate function for creating the beam file, instead of putting that logic inside another function?).

Perhaps I should try my hand at a pull request over there :-)